fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubExprAnalyzer.java

```diff
@@ -223,6 +220,15 @@ public boolean hasGroupBy() {
             return false;
         }
 
+        public boolean hasCorrelatedSlotsUnderAgg() {
+            if (!correlatedSlots.isEmpty()) {
+                return logicalPlan.anyMatch(planTreeNode -> planTreeNode instanceof LogicalAggregate
```

Comment:
this impl is inefficient. we should stop traverse when we meed the first agg node